src/main/java/io/cdap/plugin/gcp/bigquery/sink/DelegateBigQueryOutputFormat.java

```diff
@@ -0,0 +1,107 @@
+
+import com.google.cloud.hadoop.io.bigquery.output.IndirectBigQueryOutputFormat;
+import com.google.gson.JsonObject;
+import org.apache.hadoop.io.NullWritable;
+import org.apache.hadoop.mapreduce.JobContext;
+import org.apache.hadoop.mapreduce.OutputCommitter;
+import org.apache.hadoop.mapreduce.OutputFormat;
+import org.apache.hadoop.mapreduce.RecordWriter;
+import org.apache.hadoop.mapreduce.TaskAttemptContext;
+
+import java.io.IOException;
+
+/**
+ * Delegate Output Format which uses {@link IndirectBigQueryOutputFormat} as delegate. This is used to intercept
+ * IOException thrown from IndirectBigQueryOutputCommitter and add more specific error message.
+ * See CDAP-15289 for more information.
+ */
+public class DelegateBigQueryOutputFormat extends OutputFormat<JsonObject, NullWritable> {
+
+  private final OutputFormat delegate;
+
+  public DelegateBigQueryOutputFormat() {
+    this.delegate = new IndirectBigQueryOutputFormat();
+  }
+
+  @Override
+  @SuppressWarnings("unchecked")
+  public RecordWriter<JsonObject, NullWritable> getRecordWriter(TaskAttemptContext taskAttemptContext)
+    throws IOException, InterruptedException {
+    return delegate.getRecordWriter(taskAttemptContext);
+  }
+
+  @Override
+  public void checkOutputSpecs(JobContext jobContext) throws IOException, InterruptedException {
+    delegate.checkOutputSpecs(jobContext);
+  }
+
+  @Override
+  public OutputCommitter getOutputCommitter(TaskAttemptContext taskAttemptContext)
+    throws IOException, InterruptedException {
+    return new DelegateOutputCommitter(delegate.getOutputCommitter(taskAttemptContext));
+  }
+
+  /**
+   * Delegate Output committer.
+   */
+  public static class DelegateOutputCommitter extends OutputCommitter {
+    private final OutputCommitter delegate;
+
+    DelegateOutputCommitter(OutputCommitter delegate) {
+      this.delegate = delegate;
+    }
+
+    @Override
+    public void setupJob(JobContext jobContext) throws IOException {
+      delegate.setupJob(jobContext);
+    }
+
+    @Override
+    public void setupTask(TaskAttemptContext taskContext) throws IOException {
+      delegate.setupTask(taskContext);
+    }
+
+    @Override
+    public boolean needsTaskCommit(TaskAttemptContext taskContext) throws IOException {
+      return delegate.needsTaskCommit(taskContext);
+    }
+
+    @Override
+    public void commitTask(TaskAttemptContext taskContext) throws IOException {
+      delegate.commitTask(taskContext);
+    }
+
+    @Override
+    public void abortTask(TaskAttemptContext taskContext) throws IOException {
+      delegate.abortTask(taskContext);
+    }
+
+    @Override
+    public void commitJob(JobContext jobContext) throws IOException {
+      try {
+        // CDAP-15289 - add specific error message.
+        delegate.commitJob(jobContext);
+      } catch (IOException e) {
+        throw new IOException("Error occurred while writing to BigQuery. Please see BigQuery job logs for more " +
```

Comment:
the error message should contain the job id, otherwise the user won't know which logs to examine.

Reply:
The BigQuery import job is submitted by output committer. I do not think that is exposed.

Reply:
Doesn't the plugin know the job id? Without the id, this error message is not really much better than the previous one.

Reply:
Had an offline discussion with @albertshau . BigQuery outputformat `IndirectBigQueryOutputFormat` does not provide a way to get job Id. Therefore, we decided to  override logic of `IndirectBigQueryOutputCommitter#commitJob` for useful log message. This will allow us to also collect and throw underlying error message which was being truncated by `BigQueryUtils.waitForJobCompletion` method.